chitchat/src/state.rs

```diff
@@ -822,6 +773,213 @@ impl From<&ClusterState> for ClusterStateSnapshot {
     }
 }
 
+/// A thin wrapper around `NodeState` that provides a mutable view of the node state
+/// listens for key-value updates and triggers updates when necessary.
+pub struct NodeStateMut<'a> {
+    pub(crate) listeners: &'a Listeners,
+    pub(crate) node_state_mut: &'a mut NodeState,
+}
+
+impl<'a> Deref for NodeStateMut<'a> {
+    type Target = NodeState;
+
+    fn deref(&self) -> &NodeState {
+        self.node_state_mut
+    }
+}
+
+impl<'a> DerefMut for NodeStateMut<'a> {
+    fn deref_mut(&mut self) -> &mut NodeState {
+        self.node_state_mut
+    }
+}
+
```

Comment:
I think deref makes sense, but deref_mut is bound to cause mutations which don't notify listeners